### PR TITLE
Fix ChangeOutputTypeAttributes BackReplacementPattern

### PR DESCRIPTION
**Pull request by iimironov**
Ticket: 61733, 61720

Description: 
Hotfix ChangeOutputTypeAttributes for escape exceptions with fp16

### CODE REVIEW

**Review by pavel-esir — APPROVED** [review]

**Review by vgavrilo — APPROVED** [review]

File: model-optimizer/extensions/back/ChangeOutputTypeAttributes.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def find_and_replace_pattern(self, graph: Graph):
                 if node[dst_type] in [np.float32, np.float64] and ir_data_type == np.float16 and \
                         not node.has_and_set('returns_shape_value'):
                     final_type = np.float16
-                elif node.has_and_set('returns_shape_value') and node.dst_type == np.float16:
+                elif node.has_and_set('returns_shape_value') and node[dst_type] == np.float16:
```
Comment (rkazants) [inline model-optimizer/extensions/back/ChangeOutputTypeAttributes.py:55]: I think we can cover this case by unit-test. Please add it

**Review by rkazants — APPROVED** [review]

File: model-optimizer/extensions/back/ChangeOutputTypeAttributes.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ def find_and_replace_pattern(self, graph: Graph):
                 if node[dst_type] in [np.float32, np.float64] and ir_data_type == np.float16 and \
                         not node.has_and_set('returns_shape_value'):
                     final_type = np.float16
-                elif node.has_and_set('returns_shape_value') and node.dst_type == np.float16:
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/back/ChangeOutputTypeAttributes.py:55]: I don't agree with this change. The attribute "dst_type" of `Cast` operation and "output_type" of `Range` operation must always exist. Could you analyse why some models failed to convert before your change? What was the reason that the mentioned above attributes were not set?

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: model-optimizer/extensions/back/ChangeOutputTypeAttributes.py (hunk 2 above)
Comment (pavel-esir) [reply model-optimizer/extensions/back/ChangeOutputTypeAttributes.py:55]: `dst_type` is a string so we should've been referred to the node attributes as `node[dst_type]` not `node.dst_type` which is equivalent to `node."dst_type"/node."output_type"`. @iimironov's changes are correct.

**Review by pavel-esir — APPROVED** [review]

File: model-optimizer/extensions/back/ChangeOutputTypeAttributes.py (hunk 2 above)
Comment (pavel-esir) [reply model-optimizer/extensions/back/ChangeOutputTypeAttributes.py:55]: dst_type is a string so we should've been referred to the node attributes as node[dst_type] not node.dst_type which is equivalent to node."dst_type"/node."output_type". @iimironov's changes are correct.

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**